src/rust/cryptography-x509/src/certificate.rs

```diff
@@ -13,6 +13,21 @@ pub struct Certificate<'a> {
     pub signature: asn1::BitString<'a>,
 }
 
+impl Certificate<'_> {
+    /// Retrieves the extension identified by the given OID,
+    /// or None if the extension is not present (or no extensions are present).
+    pub fn get_extension(&self, oid: asn1::ObjectIdentifier) -> Option<extensions::Extension> {
+        match &self.tbs_cert.extensions {
+            None => None,
+            Some(extensions) => {
+                let mut extensions = extensions.unwrap_read().clone();
```

Comment:
NB: The `clone` here is because `unwrap_read` returns a `&SequenceOf`, which we can't then mutably borrow for the `find(...)` call.

I messed around with this a bit, but couldn't find a clean way to avoid this clone. But I might have missed something 🙂

Reply:
(I'd also like to avoid the `unwrap_read` if possible, but that appears to be a design invariant with `common::Asn1ReadableOrWritable`)

Reply:
`unwrap_read` can be avoidable once we get to an MSRV with GATs (1.65.0)

Reply:
Cool, resolving this thread then 🙂